private/bufpkg/bufimage/bufimage.go

```diff
@@ -357,10 +357,11 @@ func NewImageForProto(protoImage *imagev1.Image, options ...NewImageForProtoOpti
 	if newImageOptions.noReparse && newImageOptions.computeUnusedImports {
 		return nil, fmt.Errorf("cannot use both WithNoReparse and WithComputeUnusedImports options; they are mutually exclusive")
 	}
-	var resolver protoencoding.Resolver
+	// TODO FUTURE: right now, NewResolver sets AllowUnresolvable to true all the time
+	// we want to make this into a check, and we verify if we need this for the individual command
+	resolver := protoencoding.NewLazyResolver(protoImage.File...)
```

Comment:
This was a readability improvement (IMO). In previous iterations, `reparseImageProto` create the resolver and used it to reparse unrecognized fields. In a prior PR, I changed it to return the resolver since we want to use it from here. But I think it's easier to read to have the caller here create the resolver and pass it in to `reparseImageProto`.

Reply:
> this just optimistically creates the resolver

While true, it's a **lazy** resolver. That means it really is just allocating a placeholder and doing no other work. Only when the resolver is first used will it actually do the work to create the resolver. If creating the resolver fails at that deferred step, then all `Find...` operations will fail with the error that prevented its construction. So we aren't really doing any meaningful work eagerly -- it only happens if/when we need to _use_ the resolver.